test/tests/carousel_carousel-2-tablist.js

```diff
@@ -0,0 +1,450 @@
+  );
+
+  for (let index = tabs.length - 1; index > 0; index--) {
+
+    // Send the arrow left key to move focus
+    await tabs[index].sendKeys(Key.ARROW_LEFT);
+
+    // Check the focus is correct
+    t.true(
+      await waitAndCheckFocus(t, ex.tabSelector, index - 1),
+      'left arrow on tab "' + index + '" should put focus on the previous tab.'
+    );
+    t.true(
+      await waitAndCheckAriaSelected(t, index - 1),
+      'left arrow on tab "' + index + '" should set aria-selected="true" on previous tab.'
+    );
+    t.true(
+      await tabpanels[index - 1].isDisplayed(),
+      'left arrow on tab "' + index + '" should display the next previous panel.'
+    );
+  }
+});
+
+ariaTest('HOME key moves focus and selects tab', exampleFile, 'key-home', async (t) => {
+  t.plan(3 * ex.tabCount);
+
+  const tabs = await t.context.session.findElements(By.css(ex.tabSelector));
+  const tabpanels = await t.context.session.findElements(By.css(ex.tabpanelSelector));
+  for (let index = 0; index < tabs.length; index++) {
+
+    // Put focus on the tab
+    await openTabAtIndex(t, index);
+
+    // Send the home key to the tab
+    await tabs[index].sendKeys(Key.HOME);
+
+    // Check the focus is correct
+    t.true(
+      await waitAndCheckFocus(t, ex.tabSelector, 0),
+      'home key on tab "' + index + '" should put focus on the first tab.'
+    );
+    t.true(
+      await waitAndCheckAriaSelected(t, 0),
+      'home key on tab "' + index + '" should set aria-selected="true" on the first tab.'
+    );
+    t.true(
+      await tabpanels[0].isDisplayed(),
+      'home key on tab "' + index + '" should display the first tab.'
+    );
+  }
+});
+
+ariaTest('END key moves focus and selects tab', exampleFile, 'key-end', async (t) => {
+  t.plan(3 * ex.tabCount);
+
+  const tabs = await t.context.session.findElements(By.css(ex.tabSelector));
+  const tabpanels = await t.context.session.findElements(By.css(ex.tabpanelSelector));
+  for (let index = 0; index < tabs.length; index++) {
+
+    // Put focus on the tab
+    await openTabAtIndex(t, index);
+
+    // Send the end key to the tab
+    await tabs[index].sendKeys(Key.END);
+
+    // Check the focus is correct
+    t.true(
+      await waitAndCheckFocus(t, ex.tabSelector, tabs.length - 1),
+      'home key on tab "' + index + '" should put focus on the last tab.'
+    );
+    t.true(
+      await waitAndCheckAriaSelected(t, tabs.length - 1),
+      'home key on tab "' + index + '" should set aria-selected="true" on the last tab.'
+    );
+    t.true(
+      await tabpanels[tabs.length - 1].isDisplayed(),
+      'home key on tab "' + index + '" should display the last tab.'
+    );
+  }
+});
+
+// Keyboard interaction
+
+ariaTest('TAB moves key through buttons', exampleFile, 'rotation-key-tab', async (t) => {
+  t.plan(1);
+
+  await assertTabOrder(t, ex.allFocusableItems);
+});
+
+ariaTest('ENTER pause and start carousel motion', exampleFile, 'rotation-enter-or-space-toggle', async (t) => {
```

Comment:
There's some finicky logic around hover and focus and playing/pausing the carousel. Could there be tests added for that as well?

Reply:
We haven't ever written tests for functionality that is not specifically describe in the tables, so I don't think it's mandatory.

Reply:
This actually seems like a good place to ask about something I've wondered with the current testing setup: almost every interactive example we have has some functionality that doesn't fit neatly into any of the test ids, and yet still affects the user interaction and accessibility of the example.

So far, when I've been responsible for writing tests, I've written those tests under whichever test id seems closest (even if it's a bit of a stretch). However, we have no real way to catch their absence, and I'd love to have better guidance for how to handle test ids for these.

There's already an [issue related to this](https://github.com/w3c/aria-practices/issues/1205), but do you have any guidance in the meantime? I think this is a good case study, since it seems like there's a reasonable possibility that the hover and focus logic has some bugs that aren't being caught by the current tests.

Reply:
I don't actually have guidance -- getting in the first set of tests (for all described behavior in the tables) was such a big pushed that I didn't consider behavior outside tables :) I appreciate you bringing up the issue now. Maybe we can make an issue to cover these scenarios for this example (to let this example page move forward) and explore options in the issue you opened?

Reply:
I'd be good with that, since it's not specific to this PR 👍 

I'm still interested in whether the specific hover/focus and play pause behavior has some bugs, but I'm fine with leaving the test for that to a later issue

Reply:
We added a way to do this re: Sarah's request: https://github.com/w3c/aria-practices/pull/1227